### ISSUE TITLE
proxy BuildDefaults broken

### ISSUE THREAD

**Issue by maci0** [openshift/openshift-ansible#1938]
this is what it puts into the master-config.yaml
@sdodson I believe that is kind of wrong ;)
`
admissionConfig:
  pluginConfig:
    BuildDefaults:
      configuration:
        apiVersion: v1
        env:
        - name: HTTP_PROXY
          value: openshift_http_proxy
        - name: HTTPS_PROXY
          value: openshift_https_proxy
        - name: NO_PROXY
          value: o,p,e,n,s,h,i,f,t,_,n,o,p,r,o,x,y
        gitHTTPProxy: openshift_builddefaults_http_proxy
        gitHTTPSProxy: openshift_builddefaults_https_proxy
        kind: BuildDefaultsConfig
`

**Comment by sdodson** [comment]
@maci0 hmm, I doubt it's necessary, but what values do you have for the openshift_http_proxy, openshift_https_proxy, and openshift_no_proxy variables... or if it's easier just the whole inventory.

**Comment by maci0** [comment]
Heres the proxy portion of the inventory file. mostly taken from https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example
I think I can now see where the error is, shouldn't it be using `{{ openshift_http_proxy }}` ?
```
# Global Proxy Configuration
# These options configure HTTP_PROXY, HTTPS_PROXY, and NOPROXY environment
# variables for docker and master services.
openshift_http_proxy=http://foo.bar.se:8080
openshift_https_proxy=http://foo.bar.se:8080
openshift_no_proxy=169.254.169.254,172.31.9.5,172.0.0.0/8,xyz.foo.se,.bar.se,.localdomain,.sfa.se,.local,registry.foo.bar..se,.op.bar.se,.osp.foo.se
#
# Most environments don't require a proxy between openshift masters, nodes, and
# etcd hosts. So automatically add those hostnames to the openshift_no_proxy list.
# If all of your hosts share a common domain you may wish to disable this and
# specify that domain above.
#openshift_generate_no_proxy_hosts=True
#
# These options configure the BuildDefaults admission controller which injects
# environment variables into Builds. These values will default to their
# corresponding values above but you may set them independently. See BuildDefaults
# documentation at https://docs.openshift.org/latest/admin_guide/build_defaults_overrides.html
openshift_builddefaults_http_proxy=openshift_http_proxy
openshift_builddefaults_https_proxy=openshift_https_proxy
openshift_builddefaults_no_proxy=openshift_noproxy
openshift_builddefaults_git_http_proxy=openshift_builddefaults_http_proxy
openshift_builddefaults_git_https_proxy=openshift_builddefaults_https_proxy
```

**Comment by sdodson** [comment]
Ok, this is a case of poor documentation on my part. If you intend for the builddefaults to just copy from the others leave them commented.

**Comment by sdodson** [comment]
Can you see if this change makes things more clear? https://github.com/openshift/openshift-ansible/pull/1926

**Comment by maci0** [comment]
Ahh, ok. Sorry to bother you then :)

btw do you think that the cloudconfig auth url should also be automatically added to the no_proxy line ?

**Comment by sdodson** [comment]
@maci0 yeah, that's a good idea, i'd add that in the master's sysconfig template as I don't think any other components would need that defined, correct?

**Comment by maci0** [comment]
depends, If you have other containers talking to the cloud auth API it might be needed for docker as well. 
But that may be a corner-case. 
I also had to add the cloud metadata url to my no_proxy line here 169.254.169.254

**Comment by sdodson** [comment]
@maci0 actually, so we don't add proxy envvars to /etc/sysconfig/atomic-openshift-node so they wouldn't use a proxy by default when hitting openshift_cloudprovider_*_auth_url. Is it common to need to use a proxy for nodes to hit that url?

Aside from that I'm not sure I understand why pods themselves would need access to the cloudprovider auth url. Can you explain that usecase a bit more?

**Comment by maci0** [comment]
@sdodson at the current customer for example we have an app running on OSE that talks to OSP and stores data in swift.

But yes, its an edge case.

**Comment by sdodson** [comment]
@maci0 discussed with the S2I team. If these are S2I deployments then the BuildDefaults stuff should be copied into the final pods. If this isn't S2I then we have no way of setting Defaults or Overrides. If that's something you need then lets get an RFE in to mirror the BuildDefaults and BuildOverrides admission controllers as more generic Defaults and Overrides admission controllers.
